llvm/lib/Target/SBF/SBFInstrInfo.td

```diff
@@ -71,6 +71,21 @@ def i64immSExt32 : PatLeaf<(i64 imm),
 def i32immSExt32 : PatLeaf<(i32 imm),
                 [{return isInt<32>(N->getSExtValue()); }]>;
 
+// Fetch the upper 32-bits of a 64-bit integer.
+def Upper64 : SDNodeXForm<imm, [{
+  uint64_t value = N->getSExtValue() >> 32;
+  return CurDAG->getTargetConstant(value, SDLoc(N),
+                                   N->getValueType(0));
+}]>;
```

Comment:
Might make more sense to rename `Upper64` -> `Upper32`. Similarly for `Lower64` below.